<a href="https://colab.research.google.com/github/OrastaRakhmatullayeva/ML-projects/blob/main/integral_in_ML.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import numpy as np
import matplotlib.pyplot as plt
import sympy as sp


TASK 1

In [6]:
def f(x):
  return x**2

def left_riemann(f,a,b,n):
  dx=(b-a)/n
  xs=np.linspace(a,b,n,endpoint=False)
  return np.sum(f(xs))*dx
for n in [4,10,50]:
  print(f'n={n}:  {left_riemann(f,0,2,n):.2f} ')


n=4:  1.75 
n=10:  2.28 
n=50:  2.59 


TASK 2


 a) ∫x⁶ dx = x^7/7+c  

 b) ∫9 dx = 9*x+c

 c) ∫(4x³ + 3x²) dx =x^4+x^3+c

 d) ∫2x⁵ dx = x^6/3+c

In [12]:
x=sp.symbols("x")
F1=sp.integrate(x**6,x)
print(f'∫x^6 dx= {F1} + C')

F2=sp.integrate(9,x)
print(f'∫9 dx= {F2} + C')

F3=sp.integrate((4*x**3+3*x**2),x)
print(f'∫(4*x^3+3*x^2) dx ={F3} + C')

F4=sp.integrate((2*x**5),x)
print(f'∫2*x^5 dx = {F4} + C')

∫x^6 dx= x**7/7 + C
∫9 dx= 9*x + C
∫(4*x^3+3*x^2) dx =x**4 + x**3 + C
∫2*x^5 dx = x**6/3 + C


TASK 3

Avval F(x) antihosilasini toping, keyin F(4) - F(1):

f(x) = 3x²,  oraliq: [1, 4]
F=x^3+C

F(4)-F(1)=4^3-4^1=64-1=63

In [16]:
#f(x) = 3x²,  oraliq: [1, 4]
F=sp.integrate(3*x**2,(x,1,4))
print(f'∫3*3^2 dx= {F}')

∫3*3^2 dx= 63


TASK 4

In [17]:
def f(x):
  return x**3

def left_riemann(f, a, b, n):
    dx = (b - a) / n
    xs = np.linspace(a, b, n, endpoint=False)
    return np.sum(f(xs)) * dx

for n in [5,20]:
  print(f'n={n}:  {left_riemann(f,0,2,n):.2f} ')


n=5:  2.56 
n=20:  3.61 


In [21]:
xs = np.linspace(0, 2,6)
ys = f(xs)
natija = np.trapezoid(ys, xs)
print(f'trapetsiya (n=5) :natija: {natija}')

trapetsiya (n=5) :natija: 4.16


In [22]:
xs = np.linspace(0, 2,21)
ys = f(xs)
natija = np.trapezoid(ys, xs)
print(f'trapetsiya (n=20) :natija: {natija}')

trapetsiya (n=20) :natija: 4.01


n=5: chap=2.56,
trap=4.16 (xato: -1.44, +0.16).

n=20: chap=3.61, trap=4.01
(xato: −0.39, +0.01)

trapetsiya doim aniqroq.

TASK 5

f(x) = 4x³ + 2x

∫(4x³ + 2x) dx=x^4+x^2+C

In [26]:
F=sp.integrate((4*x**3 + 2*x),x)
print(f'∫(4x³ + 2x) dx = {F} + C')

∫(4x³ + 2x) dx = x**4 + x**2 + C


In [29]:
#checking the derivative and integral

f=4*x**3 + 2*x
print(sp.diff(F)==f)

True


TASK 6

In [30]:
#v(t) = 3t² + 1  [0,3]

t=sp.symbols('t')
V=sp.integrate((3*t**2+1),t)
print(f'∫(3t^2+1) dt = {V} + C')


∫(3t^2+1) dt = t**3 + t + C


In [32]:
V=sp.integrate((3*t**2+1),(t,0,3))
print(f'∫(3t^2+1) dt = {V}')


∫(3t^2+1) dt = -30


In [36]:
def f(x):
  return 3*x**2+1
xs = np.linspace(0, 3,10)
ys = f(xs)
natija = np.trapezoid(ys, xs)
print(f'trapetsiya (n=10) :natija: {natija:.2f}')

trapetsiya (n=10) :natija: 30.17
